contracts/src/libraries/LPMath.sol

```diff
@@ -730,7 +743,11 @@ library LPMath {
         // Newton's method will terminate as soon as the current iteration is
         // within the minimum tolerance or the maximum number of iterations has
         // been reached.
-        for (uint256 i = 0; i < SHARE_PROCEEDS_MAX_ITERATIONS; ) {
+        int256 lastDelta;
+        if (_maxIterations < SHARE_PROCEEDS_MAX_ITERATIONS) {
```

Comment:
am i incorrect in assuming that most of the time `SHARE_PROCEEDS_MAX_ITERATIONS` will be the max iterations?  if so, is there a more gas efficient implementation?

Reply:
That's correct. 

Assuming we want to verify the value, there isn't a more efficient implementation. We could just use the provided `_maxIterations`, but I find that to be sketchy since future code changes could erroneously pass a `_maxIterations` value that is smaller than `LPMath.SHARE_PROCEEDS_MAX_ITERATIONS`.